AK/String.h

```diff
@@ -196,6 +201,18 @@ class String {
         return *this;
     }
 
+    String& operator=(std::nullptr_t)
```

Comment:
Is this necessary?

Reply:
I did use that in the implementation of `InputMemoryStream& operator>>(InputMemoryStream&, String&)`. I feel like this is a useful overload, what should I do otherwise?